com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RebuildApplicationDeployModelStep.java

```diff
@@ -35,17 +33,16 @@ protected StepPhase executeStep(ExecutionWrapper execution) throws Exception {
             CloudApplicationExtended app = StepsUtil.getApp(execution.getContext());
             getStepLogger().debug(Messages.BUILDING_CLOUD_APP_MODEL, app.getName());
 
-            DeployedMta deployedMta = StepsUtil.getDeployedMta(execution.getContext());
-            List<DeployedMtaModule> deployedModules = (deployedMta != null) ? deployedMta.getModules() : Collections.emptyList();
-            Set<String> mtaArchiveModules = StepsUtil.getMtaArchiveModules(execution.getContext());
-            getStepLogger().debug(Messages.MTA_ARCHIVE_MODULES, mtaArchiveModules);
-            Set<String> deployedModuleNames = CloudModelBuilderUtil.getDeployedModuleNames(deployedModules);
-            getStepLogger().debug(Messages.DEPLOYED_MODULES, deployedModuleNames);
-            Set<String> mtaModules = StepsUtil.getMtaModules(execution.getContext());
-            getStepLogger().debug(Messages.MTA_MODULES, mtaModules);
-            // TODO: Build a cloud model only for the current application:
-            List<CloudApplicationExtended> modifiedApps = getApplicationsCloudModelBuilder(execution.getContext()).build(mtaArchiveModules,
-                mtaModules, deployedModuleNames);
+            DeploymentDescriptor descriptor = StepsUtil.getDeploymentDescriptor(execution.getContext());
+            Module applicationModule = descriptor.getModules1()
+                .stream()
+                .filter(module -> app.getModuleName()
+                    .equals(module.getName()))
+                .findFirst()
+                .orElseThrow(() -> new SLException(MessageFormat.format(
+                    "Application module name {0} is not matching any of the modules in the deployment descriptor", app.getModuleName())));
```

Comment:
Add a constant for this in cf.process.Messages.

Reply:
+1